src/test/regress/expected/failure_create_table.out

```diff
@@ -0,0 +1,597 @@
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+SELECT create_distributed_table('test_table','id');
+WARNING:  connection not open
+CONTEXT:  while executing command on localhost:57640
+ERROR:  connection error: localhost:57640
+DETAIL:  connection not open
+SELECT count(*) FROM pg_dist_shard;
+ count 
+-------
+     0
+(1 row)
+
+\c - - - :worker_1_port
+SELECT count(*) FROM pg_class WHERE relname LIKE 'test_table%';
+ count 
+-------
+     0
+(1 row)
+
+\c - - - :master_port
+SET search_path TO 'failure_create_table';
+SELECT citus.mitmproxy('conn.allow()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+-- Now, cancel the connection just after transaction is opened on
+-- workers. Note that, cancel requests will be ignored during
+-- shard creation.
+SELECT citus.mitmproxy('conn.onQuery(query="^BEGIN TRANSACTION ISOLATION LEVEL READ COMMITTED").cancel(' || pg_backend_pid() || ')');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+SELECT create_distributed_table('test_table','id');
+WARNING:  cancel requests are ignored during shard creation
+ create_distributed_table 
+--------------------------
+ 
+(1 row)
+
+SELECT count(*) FROM pg_dist_shard;
+ count 
+-------
+     4
+(1 row)
+
+\c - - - :worker_1_port
+SELECT count(*) FROM pg_class WHERE relname LIKE 'test_table%';
+ count 
+-------
+     4
+(1 row)
+
+\c - - - :master_port
+SET search_path TO 'failure_create_table';
+SELECT citus.mitmproxy('conn.allow()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+DROP TABLE test_table;
+CREATE TABLE test_table(id int, value_1 int);
+-- Kill the connection while with colocate_with option
+CREATE TABLE temp_table(id int, value_1 int);
+SELECT create_distributed_table('temp_table','id');
+ create_distributed_table 
+--------------------------
+ 
+(1 row)
+
+SELECT citus.mitmproxy('conn.onQuery(query="CREATE TABLE").kill()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+SELECT create_distributed_table('test_table','id',colocate_with=>'temp_table');
+ERROR:  server closed the connection unexpectedly
+	This probably means the server terminated abnormally
+	before or while processing the request.
+CONTEXT:  while executing command on localhost:57640
+SELECT citus.mitmproxy('conn.allow()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+SELECT count(*) FROM pg_dist_shard WHERE logicalrelid='test_table'::regclass;
+ count 
+-------
+     0
+(1 row)
+
+SELECT citus.mitmproxy('conn.onQuery(query="CREATE TABLE").cancel(' || pg_backend_pid() || ')');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+SELECT create_distributed_table('test_table','id',colocate_with=>'temp_table');
+ERROR:  canceling statement due to user request
+SELECT citus.mitmproxy('conn.allow()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+SELECT count(*) FROM pg_dist_shard WHERE logicalrelid='test_table'::regclass;
+ count 
+-------
+     0
+(1 row)
+
+DROP TABLE temp_table;
+-- Kill the connection after worker sends "PREPARE TRANSACTION" ack
```

Comment:
This comment does not match the command you sent mitmproxy. To kill at an ack you can use a string like `conn.onCommandComplete(command="PREPARE TRANSACTION")`.